src/server.js

```diff
@@ -80,7 +83,7 @@ app.use((req, res) => {
     return;
   }
 
-  match({ history, routes: getRoutes(store), location: req.originalUrl }, (error, redirectLocation, renderProps) => {
+  match({ syncedHistory, routes: getRoutes(store), location: req.originalUrl }, (error, redirectLocation, renderProps) => {
```

Comment:
Should be `history:syncedHistory`? If no history is passed, react-router will create memory history, but this new memory history won't be synced with `react-router-redux` and will cause redirects to '/' when synced on client

I'm not 100% sure, i can't test it now, but that's issue i had before